src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs

```diff
@@ -131,8 +131,16 @@ private CosmosClient CreateCosmosClientInternal(CosmosDataStoreConfiguration con
                 new CosmosClientBuilder(host, _cosmosAccessTokenProviderFactory.Invoke().TokenCredential) :
                 new CosmosClientBuilder(host, key);
 
+            if (configuration.ConnectionMode == ConnectionMode.Gateway)
```

Comment:
Should we add a logging statement to ensure that connectionMode is correctly populated from our config? If there are other means to confirm that, feel free to ignore

Reply:
In case of exceptions, we do see client configuration which displays connection mode. 
Example: "ConnectionMode":"Direct"
To save time of PR build, I am going to leave it as it is.